rest/tests/test_learning_resource.py

```diff
@@ -214,6 +245,71 @@ def test_preview_url(self):
             learning_resource.get_preview_url()
         )
 
+    def test_learning_resource_exports_invalid_methods(self):
+        """
+        Test invalid methods for session-based shopping cart
+        for learning resource exports
+        """
+
+        collection_url = (
+            "{repo_base}{repo_slug}/learning_resource_exports/".format(
+                repo_base=REPO_BASE,
+                repo_slug=self.repo.slug,
+            )
+        )
+        lr_id = LearningResource.objects.first().id
+        detail_url = (
+            "{repo_base}{repo_slug}/learning_resource_exports/{lr_id}/".format(
+                repo_base=REPO_BASE,
+                repo_slug=self.repo.slug,
+                lr_id=lr_id,
+            )
+        )
+
+        # PUT and PATCH not supported
+        resp = self.client.patch(collection_url, {})
+        self.assertEqual(HTTP_405_METHOD_NOT_ALLOWED, resp.status_code)
+        resp = self.client.put(collection_url, {})
+        self.assertEqual(HTTP_405_METHOD_NOT_ALLOWED, resp.status_code)
+        resp = self.client.patch(detail_url, {})
+        self.assertEqual(HTTP_405_METHOD_NOT_ALLOWED, resp.status_code)
+        resp = self.client.put(detail_url, {})
+        self.assertEqual(HTTP_405_METHOD_NOT_ALLOWED, resp.status_code)
+
+        # POST not supported on detail url
+        resp = self.client.post(detail_url, {})
+        self.assertEqual(HTTP_405_METHOD_NOT_ALLOWED, resp.status_code)
+
+    def test_learning_resource_exports_anonymous(self):
+        """Make sure anonymous users are forbidden"""
+        lr_id = get_resources(self.repo.id).first().id
+        self.logout()
+
+        self.get_learning_resource_export(
+            self.repo.slug, lr_id, expected_status=HTTP_403_FORBIDDEN)
+        self.get_learning_resource_exports(
+            self.repo.slug, expected_status=HTTP_403_FORBIDDEN)
+        self.create_learning_resource_export(
+            self.repo.slug, {"id": lr_id}, expected_status=HTTP_403_FORBIDDEN)
+        self.delete_learning_resource_export(
+            self.repo.slug, lr_id, expected_status=HTTP_403_FORBIDDEN)
+        self.delete_learning_resource_exports(
+            self.repo.slug, expected_status=HTTP_403_FORBIDDEN)
+
+    def test_learning_resource_exports_not_a_number(self):
+        """
+        Don't cause 500 error for non-numeric ids
+        """
+        lr_id = "notanumber"
+
+        self.get_learning_resource_export(
+            self.repo.slug, lr_id, expected_status=HTTP_404_NOT_FOUND)
+        self.create_learning_resource_export(
+            self.repo.slug, {"id": lr_id},
+            expected_status=HTTP_400_BAD_REQUEST)
+        self.delete_learning_resource_export(
+            self.repo.slug, lr_id, expected_status=HTTP_404_NOT_FOUND)
+
 
 # pylint: disable=too-many-ancestors
```

Comment:
We disable `too-many-ancestors` and `invalid-name` a fair number of times. I think we should disable them globally, because they're not the kind of things we tend to violate out of laziness or oversight, like line length and missing docstrings. Opinions, @carsongee, @pdpinch?